### DISCUSSION TITLE
How does HAFS load the slurm (or other batch system) environment on a given platform?

### DISCUSSION THREAD

**Discussion by mkavulich** [hafs-community/HAFS#298] [Q&A]
I am trying to make use of the existing logic in HAFS to load the appropriate environment on Jet for running a stand-alone job outside of the HAFS workflow. However, the include logic that I can find (`ush/hafs_pre_job.sh.inc`) does not actually load the full needed environment: notably `sbatch` can not be found in the resulting environment, so calls to `rocotorun` fail. It seems that `sbatch` gets loaded into the environment somewhere under the logic called from `rocoto/run_hafs.py`, but I can not find out where. Would someone be able to point me to where the environment to call batch system commands is loaded within the HAFS logic?

**Comment by jdkublnick** [comment]
Hello, mkavulich! Thanks for your patience; we have not forgotten about your question. We've contacted a subject matter expert on this topic for the most accurate response. As soon as we receive the information, we will follow up and provide you with an update.

**Comment by ZhanZhang-NOAA** [comment]
To load required modules on jet, try the following:
module purge
module use /apps/modules/modulefiles
module use /apps/lmod/lmod/modulefiles/Core
module use ${HOMEhafs}/modulefiles
module load hafs.jet

**Comment by gspetro-NOAA** [comment]
@mkavulich Did @ZhanZhang-NOAA 's response answer your question, or are you still looking for assistance on this?
